feat(all-clouds): add docs on cloud-init metapackages (#280)
canonical/ubuntu-cloud-docs: all-clouds/all-clouds-explanation/cloud-init-metapackages.rst

Cloud-init metapackages
=======================

> Starting on Ubuntu Oracular (24.10), `cloud-init <https://launchpad.net/ubuntu/+source/cloud-init>`

Reviewer: nit
```suggestion
Starting in Ubuntu Oracular (24.10), `cloud-init <https://launchpad.net/ubuntu/+source/cloud-init>`
```

Author: Done, thanks.

> Starting in Ubuntu Oracular (24.10), `cloud-init <https://launchpad.net/ubuntu/+source/cloud-init>`_

Reviewer 2: I think this link should just point to the main cloud-init documentation (rather than the code), since it is in the introductory paragraph and one tends to assume that the link will tell me what cloud-init is.

We should mention the source package link somewhere separately to indicate it as the location for finding the metapackages. Although, even in that link, the specific metapackages available are not obvious. So some detail about how to find them might be useful. - Maybe create a small section at the end called 'Installation guidelines' and add these details there.

Author: > I think this link should just point to the main cloud-init documentation (rather than the code), since it is in the introductory paragraph and one tends to assume that the link will tell me what cloud-init is.

> We should mention the source package link somewhere separately to indicate it as the location for finding the metapackages.

Done, thanks for the suggestion.

> Although, even in that link, the specific metapackages available are not obvious. So some detail about how to find them might be useful. - Maybe create a small section at the end called 'Installation guidelines' and add these details there.

Those packages can be found in the cloud-init source package link and find / installed using `apt-get`, `apt` and the like. But, being this page targeting Ubuntu image creators / builders, I believe your suggesting would probably be over-documenting. WDYT?

Reviewer 2: Yeah I agree, the way it is written now should be enough.

Though I had assumed those packages could be installed using apt, the package names were not evident in the source link. But we are specifying the package names in this page, so it's fine.

> added cloud-specific metapackages to better express cloud-specific dependencies and reduce image size.

Reviewer 2: I think a little explanation will be useful here. Maybe change it to something like the following (please correct any word or statement that is inaccurate, I am just trying to give an idea of the content needed):

> added some metapackages to better express cloud-specific dependencies and to reduce image size.

>The original ``cloud-init`` is still available as a metapackage and a smaller version (called ``cloud-init-base``) has been created to include just the essential dependencies. A few specific metapackages have been created, while more cloud-specific ones are under development. The current set include:

Author: Done.

> * ``cloud-init-base`` contains the cloud-init code with the minimal set of dependencies

Reviewer 2: ```suggestion
* ``cloud-init-base`` - package that contains the cloud-init code along with the minimal set of dependencies
```

Author: Done.

> * ``cloud-init`` is a metapackage that retains every previous dependency

Reviewer 2: ```suggestion
* ``cloud-init`` - metapackage that retains every previous dependency
```

Author: Done.

> * ``cloud-init-cloud-sigma`` and ``cloud-init-smart-os`` are metapackages with an

Reviewer 3: It would be helpful to have one or two examples of the cloud-specific metapackage names, along with a location where one can find the appropriate metapackage for their cloud, if such a location exists.

Author: `cloud-init-cloud-sigma` and `cloud-init-smart-os` are all the packages we plan for Oracular.

The [cloud-init source package](https://launchpad.net/ubuntu/+source/cloud-init), which is linked in the first sentence, is the canonical location to find them.

Author: Bonus: `cloud-init-azure` will come for Ubuntu PP.

Author: Note that the cloud-specific packages have not yet been implemented. This is being implemented in https://github.com/canonical/cloud-init/pull/5655.

> * ``cloud-init-cloud-sigma`` and ``cloud-init-smart-os`` are metapackages with an

Reviewer 2: ```suggestion
* ``cloud-init-cloud-sigma`` and ``cloud-init-smart-os`` - metapackages with an
```

Author: Done.

>   additional dependency on python3-serial

Reviewer 2: Add some detail here about their possible use cases. Since they are named cloud-sigma and smart-os, I am assuming they are to be used in some specific situations.

If you are an image builder:

* Install ``cloud-init``, containing all dependencies, if the target platform is unknown

> * Install a cloud specific metapackage if the target platform requires one

Reviewer 3: `cloud-specific`

Author: Done, thanks.

> * Install a cloud-specific metapackage if the target platform requires one

Reviewer 2: Move this point to the end of the list (after 'Install ``cloud-init-base`` ... )
```suggestion
* Install a cloud-specific metapackage if the target platform has some specific dependency requirements. 
```

Author: Done.

> * Install a cloud-specific cloud-init package if the target platform requires one

Reviewer: it took me a second to understand what was being conveyed here. i didn't make the connection between l11 (defining the available cloud specific packages) and this instruction. is there a better way to highlight that in the previous block to make this translate a little easier?

e.g. `... are cloud specific metapackages ...` may be enough of a tweak to make this more consumable

Author: Done, thanks.

Cloud-init metapackages
=======================

Starting in Ubuntu Oracular (24.10), `cloud-init <https://docs.cloud-init.io>`_
added `cloud-specific metapackages <https://launchpad.net/ubuntu/+source/cloud-init>`_
to better express cloud-specific dependencies and reduce image size.

The original cloud-init is still available as a metapackage and a smaller

> version (called cloud-init-base) has been created to include just the essential

Reviewer 3: Should there be backticks around cloud-init and cloud-init-base?
